### ISSUE TITLE
Error in log file: field_defs should be an array

### ISSUE THREAD

**Issue by horus68** [salesagility/SuiteCRM#5239]
#### Issue

When entering the edit view on certain modules I get this error on the suitecrm.log
`Fri Feb 16 12:53:56 2018 [6880][1][FATAL] field_defs should be an array`

#### Steps to Reproduce

Errors fired when entering edit view a record in modules:
- Account
- Opportunity
- Leads
- Tasks
- Cases

No error if editing a "contact"

#### Context

Reduce log size!

#### Your Environment

* SuiteCRM Version used: 7.10. RC2
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome
* Environment name and version (e.g. MySQL, PHP 7): apache 2.4.18 / PHP Version 7.0.11 / mysql 5.7.17
* Operating System and version (e.g Ubuntu 16.04): Windows 10 - 32 bit - local Wamp

**Comment by vmijalkovic** [comment]
Same here with AOS_Quotes editing LineItems.
error: [FATAL] field_defs should be an array
all labels: undefined

Suite 7.10
Apache 2.4.27
Windows server 2012 R2
SQL server 2016

**Comment by jsamelko** [comment]
Same again - also have this error recorded when a user logs-in and follows the 'your password has expired' process to update their PW

Suite 7.10
Apache 2.4.18
MySQL 5.7.19
PHP 7.0.24
Ubuntu 16.04 LTS

**Comment by pgorod** [comment]
@Dillon-Brown I suggest increasing the priority on this one, I am seeing this a lot in the forums lately, and always with people who have some serious problem with broken functionality.

I think the message is coming from here:
https://github.com/salesagility/SuiteCRM/blob/master/data/SugarBean.php#L542

This is reproducible in 7.10.2, I suggest either debugging or getting a debug_backtrace of that error happening, so we can see the call stack.

Finally, I suggest if anybody edits that file I linked, to make all these "should be an array" errors different from one another, so that the logs tell us immediately where it's breaking. So the error messages could be renamed to include the function name:
```
$GLOBALS['log']->fatal('populateDefaultValues: field_defs should be an array');
$GLOBALS['log']->fatal('process_union_list_query: field_defs should be an array');
etc
```
... and please fix all those `filed_defs` typos while you're there :-)

**Comment by PedroErnst** [comment]
I've been debugging some of these error messages, most of the time they happen right after a R&R, but then disappear when the page is loaded the next time.

I've made PR which makes the logging more informative, hopefully this can help track down some more serious bugs.

**Comment by pgorod** [comment]
In my tests this sometimes happens, sometimes not... I still haven't figured out a rule. But I'm pretty sure it wasn't happening until recently, the bug report says 7.10RC. I wonder if looking at the commits in that release we can find something suspicious?

**Comment by PedroErnst** [comment]
@pgorod 

The logging was added in c2988dcc6619fb1f9da49ef81a20f574518edec8, that was merged into 7.10. 

Before this the conditions could have been met, but you wouldn't have seen the logging. 
Same for 7.8 and 7.9, field_defs is sometimes not set, but it doesn't show up in the logs.

**Comment by pgorod** [comment]
@PedroErnst ah, that explains a lot :-)

But if this isn't really fatal, we should lower the logging level. These things really complicate troubleshooting, since when you don't know what problem you have, a message like this in the logs can distract your attention and send you looking for the problem where it doesn't exist.

Maybe @gymad can give us his opinion, since he made these changes? Thanks

**Comment by rsiasat00** [comment]
Findings and proposed solution for those that have encountered this error:

**SugarBean::populateDefaultValues $field_defs should be an array**

What I noticed was that the error occurs when a blank value enters the populdateDefaultValues function.

Log:

The blank data is when I logged the $this->field_defs inside the if(!is_array($this->field_defs)) condition

![image](https://user-images.githubusercontent.com/31335778/65659925-a66a2580-e05f-11e9-8588-17f3b39db060.png)

I'm not exactly sure why it's sending out blank values though but what I did then is that I added a condition to check if $this->field_defs doesn't exist / empty, just return it.

![image](https://user-images.githubusercontent.com/31335778/65660240-a6b6f080-e060-11e9-9ecb-e6a420c59a29.png)

I'm not exactly sure if there are issues with the fix but for now, I'm able to continue working on the CRM without the error showing up on my logs everytime I enter an editview page.
